### ISSUE TITLE
CustomGameModeRework

### ISSUE THREAD

**Issue by Z3rca** [TheOldRealms/TOW_Core#70]
This will separate the vanilla base game and the mod, this could mean a lot of things from not having to worry about future xml updates / TW adding more settlements ect. It also comes with CharacterCreationContent being overriden (can most likely just use the ones currently in the repo), the GameManager is overridden so we can easily remove the intro video / add our own (some kinks to this), I have built in the AutoGeneratedSaveManager see StoryMode.dll to learn more about it, the vanilla Story Mode structure (Phases) have been kept intact to make use of but the actual story has been removed to implement our own.

Also because most mods load in their mod via the Campaign/CampaignStoryMode then theoretically those mods wont affect our game (havent tested this yet).

**Comment by ShaneBadenhorst** [comment]
This will separate the vanilla base game and the mod, this could mean a lot of things from not having to worry about future xml updates / TW adding more settlements ect. It also comes with CharacterCreationContent being overriden (can most likely just use the ones currently in the repo), the GameManager is overridden so we can easily remove the intro video / add our own (some kinks to this), I have built in the AutoGeneratedSaveManager see StoryMode.dll to learn more about it, the vanilla Story Mode structure (Phases) have been kept intact to make use of but the actual story has been removed to implement our own.

Also because most mods load in their mod via the Campaign/CampaignStoryMode then theoretically those mods wont affect our game (havent tested this yet).
